### ISSUE TITLE
Ingester gives logs "write block resulted in empty block" and has a high CPU usage

### ISSUE THREAD

**Issue by wangzhao765** [cortexproject/cortex#3960]
I found that if I ingest some series for tenant A again, which have no data for several days, or upgrade configuration and rolling update ingester.  Ingester will give many logs below for several minutes and the CPU usage will go very high (about 4 times larger) at the same time. This problem will make ingester cannot pass the health check (because of the long time keeping doing something) and keep restarting while rolling update. Is that expected？
`
level=info ts=2021-03-16T02:46:40.418785141Z caller=head.go:824 org_id=<tenant> msg="Head GC completed" duration=2.967478ms
level=info ts=2021-03-16T02:46:43.949817939Z caller=compact.go:493 org_id=<tenant> msg="write block resulted in empty block"
`

**Comment by pstibrany** [comment]
Cortex 1.8 should fix this. From changelog:

```
* [ENHANCEMENT] Prometheus upgraded. #3739 #3806
  * Avoid unnecessary `runtime.GC()` during compactions.
  * Prevent compaction loop in TSDB on data gap.
```

Both of these items contributed to the problem.

**Comment by pracucci** [comment]
Should have been fixed by Cortex 1.8.
